odl/solvers/functional/default_functionals.py

```diff
@@ -2292,6 +2294,219 @@ def gradient(self):
+    """
+
+    def __init__(self, space, gamma):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        space : `FnBase`
+            Domain of the functional.
+        gamma : float
+            Smoothing parameter of Huberization. If ``gamma = 0``, then
+            functional is non-smooth corresponds to the usual L1 norm. For
+            ``gamma > 0``, it has a ``1/gamma``-Lipschitz gradient so that
+            its convex conjugate is ``gamma``-strongly convex.
+
+        Examples
+        --------
+        Example of initializing the Huber norm functional
+
+        >>> space = odl.uniform_discr(0, 1, 14)
+        >>> gamma = 0.1
+        >>> huber_norm = odl.solvers.Huber(space, gamma=0.1)
+
+        Check that if all elements are > ``gamma`` we get the L1-norm up to a
+        constant.
+
+        >>> x = 2 * gamma * space.one()
+        >>> tol = 1e-5
+        >>> constant = gamma / 2 * space.one().inner(space.one())
+        >>> l1_norm = odl.solvers.L1Norm(space) - constant
+        >>> abs(huber_norm(x) - l1_norm(x)) < tol
+        True
+
+        Check that if all elements are < ``gamma`` we get the squared L2-norm
+        times the weight 1/(2*gamma).
+
+        >>> x = gamma / 2 * space.one()
+        >>> l2_norm = 1 / (2 * gamma) * odl.solvers.L2NormSquared(space)
+        >>> abs(huber_norm(x) - l2_norm(x)) < tol
+        True
+
+        Compare Huber- and L1-norm for vanishing smoothing ``gamma=0``.
+
+        >>> x = odl.phantom.white_noise(space)
+        >>> huber_norm = odl.solvers.Huber(space, gamma=0)
+        >>> l1_norm = odl.solvers.L1Norm(space)
+        >>> abs(huber_norm(x) - l1_norm(x)) < tol
+        True
+
+        Redo previous example for a product space in two dimensions.
+
+        >>> domain = odl.uniform_discr([0, 0], [1, 1], [5, 5])
+        >>> space = odl.ProductSpace(domain, 2)
+        >>> x = odl.phantom.white_noise(space)
+        >>> huber_norm = odl.solvers.Huber(space, gamma=0)
+        >>> l1_norm = odl.solvers.GroupL1Norm(space, 2)
+        >>> abs(huber_norm(x) - l1_norm(x)) < tol
+        True
+        """
+
+        self.__gamma = float(gamma)
+        self.strong_convexity = 0
+
+        if self.gamma > 0:
+            grad_lipschitz = 1 / self.gamma
+        else:
+            grad_lipschitz = np.inf
+
+        super(Huber, self).__init__(space=space, linear=False,
+                                    grad_lipschitz=grad_lipschitz)
+
+    @property
+    def gamma(self):
+        """The smoothing parameter of the Huber norm functional."""
+        return self.__gamma
+
+    def local_norm(self, x):
+        if isinstance(self.domain, ProductSpace):
+            return PointwiseNorm(self.domain, 2)(x)
+        else:
+            return x.ufuncs.absolute()
+
+    def local_norm_operator(self):
+        if isinstance(self.domain, ProductSpace):
+            return GroupL1Norm(self.domain, 2)
+        else:
+            return L1Norm(self.domain)
```

Comment:
Sure. I find it a bit puzzling that this is needed in ODL. Why would GroupL1Norm(self.domain, 2) not work for scalar functions?

Reply:
That's a good question. Without looking at the internals, I assume it will work for product spaces with 1 component. We should think about making `GroupL1Norm` work for scalar functions as well.